pkg/operator/resources/operator/operator.go

```diff
@@ -47,20 +53,49 @@ type csvStrategySpec struct {
 	Deployments        []csvDeployments `json:"deployments"`
 }
 
-func createRole(name string) *rbacv1.Role {
+// CreateControllerRole returns role for vm-controller-operator
+func CreateControllerRole(namespace string) *rbacv1.Role {
```

Comment:
the kind of the created role resource says `ClusterRole` but the type is `rbacv1.Role`
seems like a mismatch, so if it is namespaced-scope, it shouldn't be a ClusterRole and the kind should be updated.

Reply:
you are right, will fix.

Reply:
I checked why we have it here and it seems like the logic to `client.Get` and `client.Create` rely on it. We need `ClusterRole` and `ClusterRoleBinding` but we need to handle it by generic code which works with `Deployment`.

Reply:
I'm still not sure about it - on line 61 the kind is 'ClusterRole' but the returned type of this method is a Role instead of ClusterRole.